PacketDotNet/TcpPacket.cs

```diff
@@ -421,10 +421,29 @@ public TcpPacket(ByteArraySegment bas)
                 // the length of the payload is the total payload length
                 // above, minus the length of the tcp header
                 payloadPacketOrData.TheByteArraySegment.Length = newTcpPayloadLength;
+                this.DecodePayload();
             }
         }
 
         /// <summary>
+        /// Decode Payload to Support Drda procotol
+        /// </summary>
+        /// <returns></returns>
+        public TcpPacket DecodePayload()
+        {
+            if(PayloadData==null || PayloadData.Length == 0)
+            {
+                return this;
+            }
+            if(PayloadData.Length>=10&&PayloadData[2]== 0xd0)
```

Comment:
Can you modify this to make it more obvious what is going on? For instance is 10 the minimum length of a DRDA packet that you could get from a constant in DRDAFields? I'd say the same would help clarify what 0xd0 is as well.